sygnal/gcmpushkin.py

```diff
@@ -466,5 +466,8 @@ def _build_data(n: Notification, device: Device) -> Optional[Dict[str, Any]]:
         if getattr(n, "counts", None):
             data["unread"] = n.counts.unread
             data["missed_calls"] = n.counts.missed_calls
+        
+        # this line tested for Fluffychat 
+        data["devices"] = [{"pushkey": device.pushkey,"data": device.data,"pushkey_ts": device.pushkey_ts,"app_id": device.app_id}]
```

Comment:
I don't have particular objections to adding extra fields to support fluffychat, but:

1. I think it should be configurable (in `sygnal.yaml`). There's no point adding extra data on pushes to apps that aren't going to use it.
2. I'd rather like to understand the reasons fluffychat needs this data. Shouldn't the app already know this information? (Also, why is it an array `devices` instead of a single `device`)?
3. Any changes made here will need documenting in https://github.com/matrix-org/sygnal/blob/main/docs/applications.md

Reply:
FluffyChat FCM push notification can work with that line of code,
ok, thanks a lot for your response 🙏